app/src/main/java/org/sopt/and/presentation/home/HomeViewModel.kt

```diff
@@ -1,19 +1,31 @@
 package org.sopt.and.presentation.home
 
-import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
 import dagger.hilt.android.lifecycle.HiltViewModel
-import kotlinx.coroutines.flow.MutableStateFlow
-import kotlinx.coroutines.flow.StateFlow
-import org.sopt.and.presentation.home.uistate.HomeUiState
+import kotlinx.coroutines.launch
+import org.sopt.and.presentation.home.HomeContract.HomeEvent
+import org.sopt.and.presentation.home.HomeContract.HomeSideEffect
+import org.sopt.and.presentation.home.HomeContract.HomeUiState
+import org.sopt.and.presentation.util.BaseViewModel
 import javax.inject.Inject
 
 @HiltViewModel
 class HomeViewModel @Inject constructor(
 
-) : ViewModel() {
+) : BaseViewModel<HomeUiState, HomeSideEffect, HomeEvent>() {
+    override fun createInitialState(): HomeUiState = HomeUiState()
 
-    val uiState: StateFlow<HomeUiState>
-        field = MutableStateFlow(HomeUiState())
+    override suspend fun handleEvent(event: HomeEvent) {
+        when (event) {
+            is HomeEvent.FetchMainBannersAndPosters -> setState {
+                copy(
+                    banners = dummyMainBanners,
+                    posters = dummyRecommendedPosters,
+                    rankedPosters = dummyRankedPoster
+                )
+            }
+        }
+    }
 
     // dummy data for week2, later will be replaced by API
```

Comment:
추후에 수정될 여지가 있는 코드의 주석엔 `TODO`를 달아주면 `TODO` 탭에서 관리가 가능해 유지보수가 편해집니다!
```suggestion
    // TODO: dummy data for week2, later will be replaced by API
```

Reply:
헐~ 다 고쳤다 생각했는데 남아 있었네요ㅜㅜ